.github/workflows/deploy_and_build_prod.yml

```diff
@@ -119,8 +125,73 @@ jobs:
           SENTRY_DSN: ${{ secrets.SENTRY_DSN }}
           APP_BUILD_VERSION: ${{ github.run_number }}
           APP_BINARY_VERSION: ${{ steps.latestBinaryVersion.outputs.version }}
-          DOMAIN_FORMAT: ${{ secrets.DOMAIN_FORMAT }}
-        run: expo publish --non-interactive --release-channel=${{ steps.releaseChannel.outputs.releaseChannel }}
+          DOMAIN_FORMAT: ${{ secrets.TEST_DOMAIN_FORMAT }}
+          PROJECT_ID: ${{ secrets.EXPO_PROJECT_ID }}
+        id: update
+        run: eas update --non-interactive --branch ${{ steps.releaseChannel.outputs.releaseChannel }} --message "${{ steps.getCommitMessage.outputs.commit_message }}" --json | jq '.[].id' -r > updates.txt
+  eas_build:
+    name: Build Prod
```

Comment:
unlike stg and pr, the build step is combined in prod because APP_BUILD_VERSION uses the github run number, which is a [number tied to the workflow](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables). separating it will reset the current run number